### PR TITLE
DogStatsD Data aggregation

### PR DESCRIPTION
**Pull request by l0k0ms**
### What does this PR do?

Removes a FAQ to better explain DogStatsD data aggregation logic.

### Motivation

Follow up on: https://github.com/DataDog/documentation/pull/5281

### Preview link

* https://docs-staging.datadoghq.com/gus/data-agg-dogstatsd/developers/dogstatsd/data_aggregation/

### CODE REVIEW

File: content/en/developers/faq/is-there-an-alternative-to-dogstatsd-and-the-api-to-submit-metrics-threadstats.md [hunk 1]
```diff
@@ -32,11 +37,29 @@ Threadstats is good for you if...
...
+
+### Variations
+
+* The main difference is that metrics received by Threadstats may already have a timestamp
+* Metrics are not aggregated via a centralized server, but they are aggregated and flushed in a Python thread of your script. So you'll get a per-script aggregation instead of a per host aggregation
+
+To handle timestamps, Threadstats uses 2 parameters: a flush interval and a roll-up interval.
 
-Documentation: http://datadogpy.readthedocs.org/en/latest/#datadog-threadstats-module
+* The flush interval defines the time interval between two consecutive {data aggregation + data submission}.
```
Comment (cswatt) [inline content/en/developers/faq/is-there-an-alternative-to-dogstatsd-and-the-api-to-submit-metrics-threadstats.md]: Unsure what is being said here. Is it between two consecutive data aggregation-submission instances?

File: content/en/developers/faq/is-there-an-alternative-to-dogstatsd-and-the-api-to-submit-metrics-threadstats.md [hunk 2]
```diff
@@ -32,11 +37,29 @@ Threadstats is good for you if...
...
+* The roll up interval defines the data granularity after aggregation.
 
-Source code: https://github.com/DataDog/datadogpy/blob/master/datadog/threadstats/
+### Example with flush_interval=10 and roll_up_interval=5
+
+For instance, during the flush interval of 10 seconds (between 10:00:00 and 10:00:10), Threadstats receives 5 datapoints for the same metric name (a counter) and same tags, with {timestamps, values} being:
```
Comment (cswatt) [inline content/en/developers/faq/is-there-an-alternative-to-dogstatsd-and-the-api-to-submit-metrics-threadstats.md]:
```suggestion
Consider an instance where, during a flush interval of 10 seconds (between 10:00:00 and 10:00:10), Threadstats receives 5 datapoints for the same metric name (a counter) and same tags, with the following {`timestamp`, `value`} pairs:
```

File: content/en/developers/faq/is-there-an-alternative-to-dogstatsd-and-the-api-to-submit-metrics-threadstats.md [hunk 3]
```diff
@@ -32,11 +37,29 @@ Threadstats is good for you if...
...
+
+1. {09:30:15, 1}, {10:00:00, 2}, {10:00:04,1}, {10:00:05,1}, {10:00:09,1} # 1- original datapoints
```
Comment (cswatt) [inline content/en/developers/faq/is-there-an-alternative-to-dogstatsd-and-the-api-to-submit-metrics-threadstats.md]:
```suggestion
`{09:30:15, 1}, {10:00:00, 2}, {10:00:04,1}, {10:00:05,1}, {10:00:09,1} `
```

File: content/en/developers/faq/is-there-an-alternative-to-dogstatsd-and-the-api-to-submit-metrics-threadstats.md [hunk 4]
```diff
@@ -32,11 +37,29 @@ Threadstats is good for you if...
...
+2. {09:30:10, 1}, {10:00:00, 2}, {10:00:00,1}, {10:00:05,1}, {10:00:05,1} # 2- every datapoint in the same roll_up_interval (5 seconds) gets the same timestamp
```
Comment (cswatt) [inline content/en/developers/faq/is-there-an-alternative-to-dogstatsd-and-the-api-to-submit-metrics-threadstats.md]:
```suggestion
Next, every datapoint in the same `roll_up_interval` (5 seconds) gets the same timestamp:

`{09:30:10, 1}, {10:00:00, 2}, {10:00:00,1}, {10:00:05,1}, {10:00:05,1}`

```

File: content/en/developers/faq/is-there-an-alternative-to-dogstatsd-and-the-api-to-submit-metrics-threadstats.md [hunk 5]
```diff
@@ -32,11 +37,29 @@ Threadstats is good for you if...
...
+3. {09:30:10, 1}, {10:00:00, 3}, {10:00:05,2} # 3- data is aggregated and only 4 values are eventually submitted to Datadog
```
Comment (cswatt) [inline content/en/developers/faq/is-there-an-alternative-to-dogstatsd-and-the-api-to-submit-metrics-threadstats.md]:
```suggestion
Finally, data is aggregated by timestamp. The following 4 values are then submitted to Datadog: 
`{09:30:10, 1}, {10:00:00, 3}, {10:00:05,2}`
```

File: content/en/developers/faq/is-there-an-alternative-to-dogstatsd-and-the-api-to-submit-metrics-threadstats.md (hunk 1 above)
Comment (l0k0ms) [reply content/en/developers/faq/is-there-an-alternative-to-dogstatsd-and-the-api-to-submit-metrics-threadstats.md]: indeed, prior to be sent the data goes through 2 phases: 1. aggregation  then 2. submission. The flush interval is the sum of those two phases.

File: content/en/developers/dogstatsd/data_aggregation.md [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+
+## How is aggregation performed with the DogStatsD server?
+
+[DogStatsD][2] uses a *flush interval* of 10 seconds. Every 10 seconds, [DogStatsD][2] checks all data received since the last flush (i.e., in the last 10 seconds). All values that correspond to the same metric name and the same tags are aggregated together into a single value.
+
+**Note**: With the StatsD protocol, the StatsD client doesn't send metrics with timestamps. The timestamp is added at the flush time. So for a flush occurring at 10:00:10, all data received by the [DogStatsD][2] server (embedded in the Datadog Agent) between 10:00:00 and 10:00:10 is rolled up in a single datapoint that gets 10:00:00 as timestamp.
+
+## Aggregation rules per metric type
+
+Among all values received during the same flush interval, the aggregated value send depends of the [metric type][4]:
```
Comment (olivielpeau) [inline content/en/developers/dogstatsd/data_aggregation.md]:
```suggestion
Among all values received during the same flush interval, the aggregated value send depends on the [metric type][4]:
```

File: content/en/developers/dogstatsd/data_aggregation.md [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+|----------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| [GAUGE][5]     | The latest datapoint received is sent.                                                                                                                         |
+| [COUNT][3]     | The sum of all received datapoints is sent.                                                                                                                     |
+| [HISTOGRAM][6] | The min, max, sum, avg, 95 percentiles, count, and median of all datapoints received is sent. See the [HISTOGRAM metric documentation page][7] for more details. |
+| [SET][8]       | The number of different datapoints is sent.                                                                                                                     |
+| [RATE][9]      | The value difference divided by the time difference of the last 2 datapoints received.                                                                         |
```
Comment (olivielpeau) [inline content/en/developers/dogstatsd/data_aggregation.md]: rate doesn't exist for dogstatsd metrics, so should be removed from here.

Also, there's one more metric type that we could mention here: `distribution`s, aggregated as [global distributions](https://docs.datadoghq.com/graphing/metrics/distributions/#overview)

File: content/en/developers/faq/is-there-an-alternative-to-dogstatsd-and-the-api-to-submit-metrics-threadstats.md (hunk 1 above)
Comment (cswatt) [reply content/en/developers/faq/is-there-an-alternative-to-dogstatsd-and-the-api-to-submit-metrics-threadstats.md]: Let's change it to 
```suggestion
* The flush interval defines the time interval between two consecutive data aggregation-and-submission events.
```

**Pull request merged**